charts/ingresslibchart/templates/_backendconfig.yaml

```diff
@@ -0,0 +1,22 @@
+{{- define "libchart.backendconfig.tpl" -}}
+apiVersion: cloud.google.com/v1
+kind: BackendConfig
+metadata:
+  name: {{ .Chart.Name }}-ingress-backendconfig
+  labels:
+    {{- include "helm.labels" . | nindent 4 }}
+spec:
+  timeoutSec: {{ .Values.ingress.timeoutSec }}
+  healthCheck:
+    checkIntervalSec: 5
+    timeoutSec: 5
+    healthyThreshold: 2
+    unhealthyThreshold: 2
+    type: HTTPS
+    port: 443
+    requestPath: {{ .Values.ingress.requestPath }}
+{{- end }}
+
+{{- define "libchart.backendconfig" -}}
+{{- include "libchart.util.merge" (append . "libchart.backendconfig.tpl") -}}
```

Comment:
hm, what's happening here? a bit confused why the merge is necessary!

Reply:
She said it had to do with the values file here (https://github.com/broadinstitute/terra-helm/pull/452#discussion_r692349375), but not totally sure I'm understanding this yet either -- I totally get that the merge is necessary at the call site, I'm just not totally sure about it within the chart

Reply:
From [Learning Helm](https://learning.oreilly.com/library/view/learning-helm/9781492083641/ch06.html#idm46125990551032) with the files substituted in.

> libchart.backendconfig.tpl contains a template for a resource. This will look similar to other templates. It provides a blueprint that is meant to be overridden by the caller in a chart that includes this library. libchart.backendconfig is a special template. This is the template another chart will use. It takes libchart.backendconfig.tpl along with another template, yet to be defined, containing overrides, and merges them into one output.libchart.backendconfig uses a utility function that handles the merging and is handy to reuse.

This implementation is also used in the Helm library documentation.

Reply:
I think I understand now… TIL about currying in go templates. So `libchart.util.merge` is a function of three arguments (given as a list), and `libchart.backendconfig` is a function of two arguments (also given as a list)--callers provide ["current context" and "named variable that's either empty or has overrides"](https://github.com/broadinstitute/terra-helm/pull/457/files#diff-59cf366944762a86e9b3df426e5004b7655590b3bd7a1d3ed51fefceeae942d8R1) meanwhile line 21 above provides ["defaults"](https://github.com/broadinstitute/terra-helm/pull/457/files#diff-9183a237943b4944f355c6be7ffd11e7e58b59dc1bb7054dfb0293aec30461f6R6). Sorry for not groking this, I don't have O'Reilly and haven't seen this used in the wild before.